Source/DafnyCore/AST/Types/Types.cs

```diff
@@ -182,6 +198,30 @@ public enum ExpandMode {
     return null;
   }
 
+  /// <summary>
+  /// Trim away newtypes to get to an ancestor type that either is not a newtype or is a native type.
+  /// </summary>
+  public Type TrimNewtypes() {
+    Type typ = this;
+    while (typ.AsNewtype is { NativeType: null } newtypeDecl) {
+      var subst = TypeParameter.SubstitutionMap(newtypeDecl.TypeArgs, typ.TypeArgs);
+      typ = newtypeDecl.BaseType.Subst(subst);
+    }
+    return typ.NormalizeExpand();
+  }
+
+  /// <summary>
+  /// Normalize proxies, expand synonyms, and expand unconstrained newtypes.
+  /// </summary>
+  public Type NormalizePastUnconstrainedTypes() {
```

Comment:
I don't understand why this method exists. Looking at how it's used (once), can't you call `ConstraintIsCompilable` on a `RedirectingTypeDecl` without a constraint? Shouldn't that return true only if the constraints of all its base types can be compiled?

Reply:
Good callout. I rewrote the call site and eliminated this method.